sw/device/silicon_creator/rom/bootstrap.c

```diff
@@ -44,7 +44,9 @@ hardened_bool_t bootstrap_requested(void) {
   HARDENED_CHECK_NE(bootstrap_dis, kHardenedBoolTrue);
 
   // A single read is sufficient since we expect strong pull-ups on the strap
-  // pins.
+  // pins.  We assume pinmux has already been configured (by pinmux_init) to
+  // enable the internal pull-downs on the strap pins.  As such, an external
+  // weak pull-up will not be detected as a logic-high.
   uint32_t res = launder32(kHardenedBoolTrue) ^ SW_STRAP_BOOTSTRAP;
```

Comment:
nit: I believe we also want to condition entry on POR reset. Which we may already be doing. We could add this to the comment for future reference.

Reply:
Checking the wakeup state is already being done in `rom_main`.